### DISCUSSION TITLE
Comments with relation to post or posts with relation to comments?

### DISCUSSION THREAD

**Discussion by Xyphyn** [pocketbase/pocketbase#1908] [Q&A] [answered]
I have a little app where you can post stuff. I'm planning to add a comment section. In my comments collection, should I make each comment have a relation to a post, or a post with relation to many comments? The only issue with a post with relation to many comments is that a user needs to update the post itself to edit the relation.

**Comment by ganigeorgiev** [comment, marked as answer]
It will be better and easier to setup the Collection API rules if the comments have a single `relation` field to the `posts` (this also guarantees that a single comment is linked only to 1 `posts` record).

If you go with the other approach (aka. have a multiple-`relation` field in the `posts` collection), then every user who is able to submit a comment will be also able to modify the already submitted comment relations, because when you perform update in this case you update the entire relation array (aka. you'll submit `["commentId1", "commentId2", ...]`).

**Reply by Xyphyn** [reply]
Cool, thanks! Should I do the same thing for a like count? (have a separate `likes` collection and add entries)?

**Reply by ganigeorgiev** [reply]
@Xyphyn I'm not sure that I understand the use case, but yes, if it is similar to the comments, then I'd recommend a separate collection with `relation` field pointing to the main one.

The multi-`relation` field is useful in scenarios where you have a many-many or a common collection that is referenced from multiple other places, for example `labels`, `tags`, `categories`, etc. In a traditional SQL setup, for such cases you usually would need a junction table (eg. `postTagRels`) that will hold only the references between the linked entries, but PocketBase allows you to skip the 3rd table and directly hold the reference as an array (this of course is not always what you may want and sometimes the traditional approach with the junction collection could be better).
